### PR TITLE
Hypertable v1 with partial selection

### PR DESCRIPTION
**Pull request by JulienVannier66**
### What does this PR do?

Hypertable v1 with partial selection:
- Add selection component
- Change logic for selection
- Update less var to css var

Related to : https://github.com/upfluence/backlog/issues/2318

### What are the observable changes?
Peek 2023-02-01 16-57

### 🧑‍💻 Developer Heads Up

⚡ Since we are using [Ember Octane](https://blog.emberjs.com/octane-is-here/) now:
* Feel free to migrate existing components to Glimmer Components.
* Write new ones exclusively in it.

Useful Resource : [Ember Octane vs Classic Cheat Sheet](https://ember-learn.github.io/ember-octane-vs-classic-cheat-sheet/)

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/hyper-table/index.hbs [hunk 1]
```diff
@@ -1,125 +1,129 @@
...
+          {{#if (eq this._searchComponentType "string")}}
+            {{component this.searchComponent manager=this.manager}}
+          {{else}}
+            {{this.searchComponent}}
+          {{/if}}
         {{else}}
-          {{this.searchComponent}}
+          <Input
+            @type="text" @placeholder={{this.placeholder}} @value={{this._searchQuery}}
+            data-control-name="table_search_input" class="form-control upf-input" />
```
Comment (Miexil) [inline addon/components/hyper-table/index.hbs:20]: Not sure what really changed at this level, but while you're in there, maybe you can replace the default Ember input with ours ?
```
<OSS::InputContainer @placeholder={{this.placeholder}} @value={{this._searchQuery}} data-control-name="table_search_input" />
```

Unless we don't have access to OSS here 🤔 I can't remember 🤷

File: addon/components/hyper-table/index.hbs [hunk 2]
```diff
@@ -1,125 +1,129 @@
...
-            @type="text" @value={{this._availableFieldsKeyword}}
-            data-control-name="field_search_input" class="upf-input"
-            placeholder={{t "hypertable.features.manage_fields.search_placeholder"}}
-          />
+          {{#each this._fieldCategories as |category|}}
+            <div
+              class="field-category {{if (eq this._activeFieldCategory category.key) 'field-category--active'}}"
+              data-control-name={{concat "field_category_toggle_" category.key}}
+              {{action "setFieldCategory" category.key}}>
+              <div>{{category.label}}</div>
```
Comment (Miexil) [inline addon/components/hyper-table/index.hbs:77]: this.category.label or @category.label ?

File: translations/en-us.yaml [hunk 3]
```diff
@@ -75,6 +75,6 @@ hypertable:
   selection:
     creator_selected: '{count, plural, =0 {{count} creator is selected} =1 {{count} creator is selected} other {{count} creators are selected}}'
     all_creators_selected: |
-      {count, plural, =1 {All <b>{count}</b> campaign creator is selected} other {All <b>{count}</b> campaign creators are selected}}
-    select_all: '{count, plural, =1 {Select all {count} campaign creator} other {Select all {count} campaign creators}}'
+      {count, plural, =1 {All <b>{count}</b> creator is selected} other {All <b>{count}</b> creators are selected}}
```
Comment (Miexil) [inline translations/en-us.yaml:78]: Let's discuss this with Simon. :)

**Review by Miexil — APPROVED** [review]

File: addon/components/hyper-table/index.hbs (hunk 1 above)
Comment (JulienVannier66) [reply addon/components/hyper-table/index.hbs:20]: Yes we have access to OSS here, it was legacy code, I can try to update it :smile:

File: addon/components/hyper-table/index.hbs (hunk 2 above)
Comment (JulienVannier66) [reply addon/components/hyper-table/index.hbs:77]: Neither of two, it comes from the following code `{{#each this._fieldCategories as |category|}}`

File: addon/components/hyper-table/index.hbs [hunk 4]
```diff
@@ -1,125 +1,129 @@
...
-    {{#if this.manager.options.features.filtering}}
-      <OSS::Button
-        @label={{t "hypertable.features.filtering.reset_all"}}
-        class="margin-right-px-12 margin-bottom-px-6"
-        data-control-name="table_reset_filters"
-        {{action "resetAllFilters"}} />
-    {{/if}}
+      {{#if this.manager.options.features.tableViews}}
+        <OSS::Button
+          @icon="fa fa-eye"
```
Comment (phndiaye) [inline addon/components/hyper-table/index.hbs:44]: `far fa-eye` ?

File: app/styles/cells.less [hunk 5]
```diff
@@ -56,12 +48,12 @@
       }
     }
     &.empty-value {
-      color: @color-text-lighter;
+      color: var(--color-gray-300);
```
Comment (phndiaye) [inline app/styles/cells.less:51]: `500` instead ? as in the templates, we would have used `font-color-gray-500` for an empty state

File: app/styles/cells.less [hunk 6]
```diff
@@ -138,8 +130,8 @@
     display: flex;
     align-items: center;
     justify-content: space-between;
-    padding: @spacing-xxx-sm @spacing-xx-sm;
-    border-radius: @default-radius;
+    padding: var(--spacing-px-6) var(--spacing-px-6);
```
Comment (phndiaye) [inline app/styles/cells.less:133]: `padding: var(--spacing-px-6)` just, as it's the same value for all directions

File: app/styles/cells.less [hunk 7]
```diff
@@ -157,17 +149,13 @@
     }
 
     &.tether-enabled {
-      border: 1px solid lighten(@upf-gray, 15%);
+      border: 1px solid var(--color-gray-200);
```
Comment (phndiaye) [inline app/styles/cells.less:152]: `var(--border-color-default)` instead ?

File: app/styles/cells.less [hunk 8]
```diff
@@ -176,13 +164,13 @@
...
-    border-color: @upf-primary-bright-purple;
+    border-color: var(--color-primary-500);
   }
 
   .is-invalid > .form-control {
-    border-color: @upf-feedback-error;
+    border-color: var(--color-error-600);
     &:focus {
-      border-color: @upf-feedback-error;
+      border-color: var(--color-error-600);
```
Comment (phndiaye) [inline app/styles/cells.less:173]: `500` no ? 🤔

File: app/styles/cells.less [hunk 9]
```diff
@@ -295,7 +282,7 @@
   padding-top: 0;
   padding-bottom: 0;
   border: 0;
-  background-color: @upf-feedback-error !important;
+  background-color: var(--color-error-600) !important;
```
Comment (phndiaye) [inline app/styles/cells.less:285]: same as above? why not just `500`

**Review by phndiaye — APPROVED** [review]

**Pull request merged**